Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3194,6 +3195,12 @@ extension BrowserViewController: PreferencesObserver {
       PrivacyReportsManager.scheduleNotification(debugMode: !AppConstants.buildChannel.isPublic)
     case Preferences.PrivacyReports.captureVPNAlerts.key:
       PrivacyReportsManager.scheduleVPNAlertsTask()
+    case Preferences.Wallet.defaultWallet.key:
+      tabManager.reset()
+      tabManager.reloadSelectedTab()
+      for tab in self.tabManager.allTabs where tab != self.tabManager.selectedTab {
+        tab.deleteWebView()
+      }
```

Comment:
comment on my own code lol but i think we should deleteWebView only `defaultWallet` become `.none`

Reply:
`tabManager.reset()` calls `resetWebView()` which then calls `deleteWebView()` internally, so this loop shouldn't be needed.

Reply:
ah good catch.